README.md

```diff
@@ -15,6 +15,193 @@ Documentation
 
 Documentation is automatically generated from source code comments and rendered as a static website hosted via GitHub Pages at:  https://yml-org.github.io/YCoreUI/
 
+Usage
+----------
+
+### 1. UIView extensions for declarative Auto Layout
+
+To aid in auto layout, Y—CoreUI has several `UIView` extensions that simplify creating layout constraints. These do not use any 3rd party library such as SnapKit, but are simply wrappers around Apple’s own `NSLayoutConstraint` api’s. If you are more comfortable using Apple’s layout constraint api’s natively, then by all means go ahead and use them. But these convenience methods allow for less verbose code that expresses its intent more directly.
+
+All the extensions are to `UIView` and begin with the word `constrain`.
+
+The simplest flavor just creates constraints using attributes just like the original iOS 6 `NSLayoutContraint` api.
+
+```swift
+// constrain a button's width to 100
+let button = UIButton()
+addSubview(button)
+button.constrain(.width, constant: 100)
+
+// constrain view to superview
+let container = UIView()
+addSubview(container)
+container.constrain(.leading, to: .leading, of: superview)
+container.constrain(.trailing, to: .trailing, of: superview)
+container.constrain(.top, to: .top, of: superview)
+container.constrain(.bottom, to: .bottom, of: superview)
+```
+
+Another flavor creates constraints using anchors just like the anchor api’s first introduced in iOS 9.
+
+```swift
+// constrain a button's width to 100
+ let button = UIButton()
+ addSubview(button)
+ button.constrain(.widthAnchor, constant: 100)
+
+// constrain view to superview
+ let container = UIView()
+ addSubview(container)
+ container.constrain(.leadingAnchor, to: leadingAnchor)
+ container.constrain(.trailingAnchor, to: trailingAnchor)
+ container.constrain(.topAnchor, to: topAnchor)
+ container.constrain(.bottomAnchor, to: bottomAnchor)
+ ```
```

Comment:
```suggestion
let button = UIButton()
addSubview(button)
button.constrain(.widthAnchor, constant: 100)

// constrain view to superview
let container = UIView()
addSubview(container)
container.constrain(.leadingAnchor, to: leadingAnchor)
container.constrain(.trailingAnchor, to: trailingAnchor)
container.constrain(.topAnchor, to: topAnchor)
container.constrain(.bottomAnchor, to: bottomAnchor)
```
Somehow just noticed that a stray space slipped into this code block ahead of each line of code (but not the comments)